prov/efa/src/rdm/efa_rdm_cq.h

```diff
@@ -20,4 +21,20 @@ int efa_rdm_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 void efa_rdm_cq_poll_ibv_cq_closing_ep(struct efa_ibv_cq *ibv_cq, struct efa_rdm_ep *closing_ep);
 int efa_rdm_cq_poll_ibv_cq(ssize_t cqe_to_process, struct efa_ibv_cq *ibv_cq);
 
+static inline struct efa_rdm_pke *efa_rdm_cq_get_pke_from_wr_id(struct efa_ibv_cq *ibv_cq, uint64_t wr_id)
```

Comment:
if you remove the `ibv_cq` argument of this function, and move the wc unsolicited check outside, like:

```
#if ENABLE_DEBUG
static inline
struct efa_rdm_cq_get_pke_from_wr_id(uint64_t wr_id)
{
        struct efa_rdm_pke *pkt_entry;
		uint8_t gen = wr_id & (EFA_RDM_BUFPOOL_ALIGNMENT - 1);
		wr_id &= ~(EFA_RDM_BUFPOOL_ALIGNMENT - 1);
		pkt_entry = (struct efa_rdm_pke *) wr_id;
		assert(pkt_entry->gen == gen);
}
#endif

// ... in efa_rdm_cq_process_wc_*
pkt_entry = (struct efa_rdm_pke *) wr_id;
#if ENABLE_DEBUG
	if (!efa_cq_wc_is_unsolicited(ibv_cq))
		pkt_entry = efa_rdm_cq_get_pke_from_wr_id(wr_id);
#endif
```

You can make the function reused by both both code and unit test.

Reply:
sure, that works too